### PR TITLE
refactor: update error management and config/process

### PR DESCRIPTION
**Pull request by Freyskeyd**
# Description

This PR updates how we deal with config and errors during the start of the node.
Basically, removing and refactoring `Config` management and responsibility of parsing, removing some `Box<dyn Error>` to proper error handling.

## PR Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

**Review by hadjiszs — APPROVED** [review]

File: crates/topos-config/src/node.rs [hunk 1]
```diff
@@ -26,31 +28,97 @@ pub struct NodeConfig {
...
+        Self::build_config(node_path, home_path, config)
+    }
+
+    /// Try to create a new node config struct from the given home path and node name.
+    ///
+    /// It doesn't check the existence of the config file.
+    /// It's useful for creating a config file for a new node, relying on the default values.
+    pub fn new_unchecked<S: Serialize>(
+        home_path: &Path,
+        name: &str,
```
Comment (dvdplm) [inline crates/topos-config/src/node.rs:73]:
```suggestion
        node_folder: &str,
```

File: crates/topos-config/src/node.rs [hunk 2]
```diff
@@ -26,31 +28,97 @@ pub struct NodeConfig {
...
+        }
+
+        Self::build_config(node_path, home_path, config)
+    }
+
+    /// Try to create a new node config struct from the given home path and node name.
+    ///
+    /// It doesn't check the existence of the config file.
+    /// It's useful for creating a config file for a new node, relying on the default values.
+    pub fn new_unchecked<S: Serialize>(
```
[the remark this answers is not in the document]
Comment (dvdplm) [reply crates/topos-config/src/node.rs:71]: `unchecked` is commonly used for unsafe versions. How about `try_create` or `try_new`? I'd also be ok making `build_config` public and remove this method.

File: crates/topos-config/src/node.rs [hunk 3]
```diff
@@ -71,3 +139,18 @@ impl Config for NodeConfig {
         "default".to_string()
     }
 }
+
+impl AsRef<NodeConfig> for NodeConfig {
+    fn as_ref(&self) -> &Self {
+        self
+    }
+}
```
Comment (dvdplm) [inline crates/topos-config/src/node.rs:147]: Is this here to make `from` work for both by-ref and by-value?

File: crates/topos/tests/config.rs [hunk 4]
```diff
@@ -147,12 +147,14 @@ mod serial_integration {
     }
 
     /// Test node init env arguments
+    #[rstest]
     #[tokio::test]
-    async fn command_init_precedence_env() -> Result<(), Box<dyn std::error::Error>> {
-        let tmp_home_directory = tempdir()?;
-
+    async fn command_init_precedence_env(
+        create_folder: PathBuf,
```
[the remark this answers is not in the document]
Comment (dvdplm) [reply crates/topos/tests/config.rs:153]: Is `rtest` rewriting the function arguments into `FnOnce` or something? Scary!

File: crates/topos/tests/config.rs [hunk 5]
```diff
@@ -147,12 +147,14 @@ mod serial_integration {
...
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        let tmp_home_directory = create_folder;
         // Test node init with env variables
-        let node_init_home_env = tmp_home_directory.path().to_str().unwrap();
+        let node_init_home_env = tmp_home_directory.to_str().unwrap();
```
Comment (dvdplm) [inline crates/topos/tests/config.rs:157]:
```suggestion
        let node_init_home_env = tmp_home_directory.to_str().expect("path names are valid utf-8");
```

(they're not, but that it the expectation here)

Comment (dvdplm) [reply crates/topos/tests/config.rs:157]: If we're in here cleaning things up, can't we return an error here instead of panicking?

**Review by dvdplm — APPROVED** [review]

File: crates/topos/tests/config.rs (hunk 5 above)
Comment (Freyskeyd) [reply crates/topos/tests/config.rs:157]: This is tests, I'm ok to let the panic

File: crates/topos-config/src/node.rs (hunk 1 above)
Comment (Freyskeyd) [reply crates/topos-config/src/node.rs:73]: I will rename to `node_name` as it is the expected parameter, if it was a `PathBuf` `node_folder` would work

**Pull request merged**